docs/topics/typecasts.md

```diff
@@ -110,27 +255,29 @@ Smart casts can be used in the following conditions:
 
 ## "Unsafe" cast operator
 
-Usually, the cast operator throws an exception if the cast isn't possible. Thus, it's called _unsafe_.
-The unsafe cast in Kotlin is done by the infix operator `as`.
+To explicitly cast an object to a non-nullable type, use the *unsafe* cast operator `as`:
 
 ```kotlin
 val x: String = y as String
 ```
 
-Note that `null` cannot be cast to `String`, as this type is not [nullable](null-safety.md).
-If `y` is null, the code above throws an exception.
-To make code like this correct for null values, use the nullable type on the right-hand side of the cast:
+If the cast isn't possible, the compiler throws an exception. This is why it's called _unsafe_.
+
+In the previous example, if `y` is `null`, the code above also throws an exception. This is because `null` can't be cast to `String`, as `null` isn't [nullable](null-safety.md).
+To make the example work for possible null values, use a nullable type on the right-hand side of the cast:
 
 ```kotlin
 val x: String? = y as String?
 ```
 
 ## "Safe" (nullable) cast operator
 
-To avoid exceptions, use the *safe* cast operator `as?`, which returns `null` on failure.
+To explicitly cast an object to a nullable type, use the *safe* cast operator `as?`:
```

Comment:
"explicitly cast an object to a nullable type" is technically incorrect

`y as String?` - cast to a nullable type
`y as? String` - it's just a safe-cast

The difference is that if `y` has an unrelated type (e.g. `Int`), safe-cast will return `null`, but cast to nullable will throw an Exception

Reply:
As discussed, reverted to previous text.